horovod/mxnet/__init__.py

```diff
@@ -68,6 +68,17 @@ def set_wd_mult(self, args_wd_mult):
         self._optimizer.set_wd_mult(args_wd_mult)
 
 
+class DistributedTrainer(mx.gluon.Trainer):
+    def __init__(self, params, optimizer, optimizer_params=None):
+        super(DistributedTrainer, self).__init__(
```

Comment:
Do you think it would be a good idea to add something here to deal with the case where a user erroneously passes a `DistributedOptimizer` into the `DistributedTrainer`? You could just automatically unwrap the `DistributedOptimizer` for them (and maybe print a warning message).

Reply:
Make sense. Will add the check.